core/app/models/spree/variant.rb

```diff
@@ -321,7 +321,19 @@ def should_track_inventory?
     #   not from this variant
     # @return [Spree::Image] the image to display
     def display_image(fallback: true)
-      images.first || (fallback && product.variant_images.first) || Spree::Image.new
+      image_list = applicable_variant_image_rule.nil? ? images : applicable_variant_image_rule.images
+      image_list.first || (fallback && product.variant_images.first) || Spree::Image.new
+    end
+
+    # This method is meant to be used while the direct association between
+    # images and variants is being deprecated (in favor of variant image
+    # rules). This method will favor images determined by the variant image
+    # rules but will fallback to the associated images when none are present
+    # for the sake of backwards compatability.
+    #
+    # @return [Spree::Image] images associated with the variant
+    def display_images
+      applicable_variant_image_rule.try(:images) || images
```

Comment:
The `||` is intentional, I only want to load the associated images if there are no images determined by the rules.